standards.md

```diff
@@ -1,47 +1,126 @@
-# Tekton Development Standards
+# Tekton Pipelines Contributor and Reviewer Expectations
 
-This doc contains the standards we try to uphold across all project
-in the Tekton org.
+The purpose of this doc is to:
 
-## Principles
+* Outline for contributors what we expect to see in a pull request
+* Establish a baseline for reviewers so they know at a minimum what to look for
 
-When possbile, try to practice:
+Each Pull Request is expected to meet the following expectations around:
 
-- **Documentation driven development** - Before implementing anything, write
-  docs to explain how it will work
-- **Test driven development** - Before implementing anything, write tests to
-  cover it
+* [Pull Request Description](#pull-request-description)
+* [Commits](#commits)
+* [Docs](#docs)
+* [Tests](#tests)
+* [API Design](#api-design)
+* [Functionality](#functionality)
+* [Content](#content)
+* [Code](#code)
 
-Minimize the number of integration tests written and maximize the unit tests!
-Unit test coverage should increase or stay the same with every PR.
+_See also [the Tekton review process](https://github.com/tektoncd/community/blob/master/process.md#reviews)._
 
-This means that most PRs should include both:
+## Pull request description
 
-1. Tests
-2. Documentation updates
+* Include a link to the issue being addressed, but describe the context for the reviewer
+  * If there is no issue, consider whether there should be one:
+    * New functionality must be designed and approved, may require a TEP
+    * Bugs should be reported in detail
+  * Release notes filled in for user visible changes (bugs + features),
+    or removed if not applicable (refactoring, updating tests)
+  * If the template contains a checklist, it should be checked off
 
-## Commit Messages
+## Commits
 
-All commit messages should follow
-[these best practices](https://chris.beams.io/posts/git-commit/), specifically:
+* Follow [commit messages best practices](https://chris.beams.io/posts/git-commit/):
+  1. Separate subject from body with a blank line
+  2. Limit the subject line to 50 characters
```

Comment:
I don't think I've been great about following this one myself. How strongly do we care? Should we enforce this in CI?

Reply:
I agree, it would be a good use of the reviewers time if this can be automated.

Reply:
I am 💯  for enforcing this in CI (and any of the other items in this list as well!)

tbh I'm a bit surprised that github doesn't flag it, but to answer your question about why this is important, its mostly b/c tools (like github!) usually treat shorter subject lines better.

for example:

GitHub defaults to the commit message subject as the PR title and truncates it when too long (sorry @dlorenc for singling you out here):

![image](https://user-images.githubusercontent.com/432502/89324838-e756a180-d655-11ea-863a-ff4cb1c1797c.png)